src/Apps/Collect/Routes/Collection/Components/Header/DefaultHeaderArtwork.tsx

```diff
@@ -0,0 +1,87 @@
+import { Image } from "@artsy/palette"
+import { DefaultHeaderArtwork_artwork } from "__generated__/DefaultHeaderArtwork_artwork.graphql"
+import { AnalyticsSchema } from "Artsy/Analytics"
+import { useTracking } from "Artsy/Analytics/useTracking"
+import React, { useCallback } from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import styled from "styled-components"
+
+const Link = styled.a`
+  display: flex;
+  align-items: flex-end;
+`
+
+export type DefaultHeaderArtworkProps = React.HTMLAttributes<
+  HTMLAnchorElement
+> & {
+  artwork: DefaultHeaderArtwork_artwork
+  small?: boolean
+  collectionId: string
+  collectionSlug: string
+}
+
+export const DefaultHeaderArtwork: React.FC<DefaultHeaderArtworkProps> = ({
+  artwork,
+  small = false,
+  collectionId,
+  collectionSlug,
+  ...rest
+}) => {
+  const { trackEvent } = useTracking()
+
+  const handleClick = useCallback(() => {
+    trackEvent({
+      action_type: AnalyticsSchema.ActionType.Click,
+      context_module: AnalyticsSchema.ContextModule.ArtworkBanner,
+      context_page_owner_type: AnalyticsSchema.OwnerType.Collection,
+      context_page: AnalyticsSchema.PageName.CollectionPage,
+      context_page_owner_id: collectionId,
+      context_page_owner_slug: collectionSlug,
+      destination_path: artwork.node.href,
+    })
+  }, [artwork, trackEvent])
```

Comment:
@dzucconi - I've noticed this in a lot of your other work, but I was wondering if you could give a couple words on when to use `useCallback`. I've always thought about it as an opt-in for performance optimization of slow code, but I could also see it being used all the time, so that later optimizations don't need to take place. Wondering what yr thinking is.

Reply:
I got in the habit of just using `useMemo` and `useCallback` everywhere just by default. But I think that's more to do with my mental model of hooks which might just be plain wrong. https://kentcdodds.com/blog/usememo-and-usecallback 🤔

Reply:
This paragraph from that link is what I tend to think about until I've experientially encountered slow code:

> Specifically the cost for useCallback and useMemo are that you make the code more complex for your co-workers, you could make a mistake in the dependencies array, and you're potentially making performance worse by invoking the built-in hooks and preventing dependencies and memoized values from being garbage collected. Those are all fine costs to incur if you get the performance benefits necessary, but it's best to measure first.